packages/_shared/src/utils/entityHelpers.ts

```diff
@@ -199,10 +199,53 @@ export function getEntryTitle({
   return titleOrDefault(title, defaultTitle);
 }
 
-export function getEntryStatus(sys: Entry['sys']) {
+type AsyncPublishStatus = 'draft' | 'published' | 'changed';
+
+type FieldStatus = {
+  '*': Record<string, AsyncPublishStatus>;
+};
+
+export function getEntryStatus(
+  //TODO: remove union after fieldStatus is added to App SDK
+  sys: Entry['sys'] & { fieldStatus?: FieldStatus },
+  locales?: LocaleProps[]
```

Comment:
NIT: could make it more flexible if you only want to provide one locale
```suggestion
  locale?: LocaleProps | LocaleProps[]
```